### PR TITLE
Benchmark PlotMethod3 in all frames

### PR DESCRIPTION
**Pull request by eggrobin**
```
> .\Release\x64\benchmarks.exe --benchmark_filter=Planetarium
2023-06-10T13:32:01+02:00
Running C:\Users\robin\Projects\mockingbirdnest\Principia\Release\x64\benchmarks.exe
Run on (28 X 3312 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x14)
  L1 Instruction 32 KiB (x14)
  L2 Unified 1024 KiB (x14)
  L3 Unified 19712 KiB (x1)
--------------------------------------------------------------------------------------------------
Benchmark                                                        Time             CPU   Iterations
--------------------------------------------------------------------------------------------------
BM_PlanetariumPlotMethod3/NearPolarPerspectiveECI             17.0 ms         16.7 ms           45 16211 points within -1.49365 ± 7028.58 × 3.70166 ± 7029.11 × -0.0446205 ± 77.5489
BM_PlanetariumPlotMethod3/FarPolarPerspectiveECI              6.65 ms         6.88 ms           75 7408 points within -1.36987 ± 7028.41 × 3.62988 ± 7029.06 × -0.0498848 ± 77.5392
BM_PlanetariumPlotMethod3/NearEquatorialPerspectiveECI        19.5 ms         19.8 ms           34 20150 points within -1.51196 ± 7028.61 × 4.06958 ± 7028.77 × -0.0441666 ± 77.5434
BM_PlanetariumPlotMethod3/FarEquatorialPerspectiveECI         5.37 ms         5.14 ms          146 5915 points within -1.83716 ± 7026.9 × 14.916 ± 7016.15 × -0.0450745 ± 77.5429
BM_PlanetariumPlotMethod3/NearPolarPerspectiveECEF            1.32 ms         1.33 ms          448 1406 points within 4623.84 ± 315.929 × 5279.21 ± 274.064 × -0.0604935 ± 77.2085
BM_PlanetariumPlotMethod3/FarPolarPerspectiveECEF            0.129 ms        0.129 ms         4978 202 points within 4624.78 ± 314.986 × 5279.21 ± 274.064 × -0.903881 ± 73.813
BM_PlanetariumPlotMethod3/NearEquatorialPerspectiveECEF       1.91 ms         1.93 ms          373 2256 points within 4623.84 ± 315.926 × 5279.21 ± 274.064 × -0.13179 ± 77.4645
BM_PlanetariumPlotMethod3/FarEquatorialPerspectiveECEF       0.718 ms        0.711 ms         1120 678 points within 4623.86 ± 315.912 × 5279.21 ± 274.064 × 0.57795 ± 76.9242
BM_PlanetariumPlotMethod3/NearPolarPerspectiveGSE             81.0 ms         82.4 ms           11 16317 points within -1.2998 ± 7027.21 × 3.84058 ± 7027.79 × -7.52759 ± 4127.34
BM_PlanetariumPlotMethod3/FarPolarPerspectiveGSE              40.8 ms         38.6 ms           17 7260 points within -1.17798 ± 7026.31 × 3.85425 ± 7027.78 × -6.96802 ± 4127.02
BM_PlanetariumPlotMethod3/NearEquatorialPerspectiveGSE        94.0 ms         96.0 ms            7 19359 points within -1.32495 ± 7027.17 × 4.62378 ± 7027.05 × -7.25586 ± 4127.08
BM_PlanetariumPlotMethod3/FarEquatorialPerspectiveGSE         36.7 ms         36.2 ms           19 6011 points within -1.62256 ± 7025.46 × 14.8474 ± 7014.97 × -8.79053 ± 4125.58
BM_PlanetariumPlotMethod3/NearPolarPerspectiveEML             91.7 ms         92.0 ms            9 15856 points within -0.643555 ± 7939.5 × 5.6582 ± 7241.7 × -6.72559 ± 5285.31
BM_PlanetariumPlotMethod3/FarPolarPerspectiveEML              45.4 ms         45.8 ms           15 7113 points within -0.655762 ± 7937.67 × 4.70337 ± 7241.12 × -5.95703 ± 5280.89
BM_PlanetariumPlotMethod3/NearEquatorialPerspectiveEML         104 ms          102 ms            6 17845 points within -0.211182 ± 7940.2 × 5.56128 ± 7242.08 × -6.86597 ± 5285.43
BM_PlanetariumPlotMethod3/FarEquatorialPerspectiveEML         42.4 ms         42.3 ms           17 6230 points within 0.322021 ± 7939.67 × 5.87646 ± 7241.74 × -6.2478 ± 5282.72
BM_PlanetariumPlotMethod3/NearPolarPerspectiveSEL              186 ms          184 ms            4 16376 points within -97.2134 ± 7749.1 × -715.116 ± 7746.46 × -338.572 ± 4223.46
BM_PlanetariumPlotMethod3/FarPolarPerspectiveSEL              86.7 ms         82.6 ms            7 7217 points within -97.8821 ± 7748.45 × -715.6 ± 7745.42 × -339.1 ± 4221.32
BM_PlanetariumPlotMethod3/NearEquatorialPerspectiveSEL         220 ms          219 ms            3 19226 points within -96.8335 ± 7749.81 × -712.32 ± 7743.61 × -338.678 ± 4223.48
BM_PlanetariumPlotMethod3/FarEquatorialPerspectiveSEL         79.4 ms         78.1 ms            9 5956 points within -97.0276 ± 7748.75 × -716.374 ± 7742.94 × -338.613 ± 4223.4
```

### CODE REVIEW

File: benchmarks/planetarium_plot_methods.cpp [hunk 1]
```diff
@@ -232,32 +324,43 @@ void RunBenchmark(benchmark::State& state,
...
+      ->Unit(benchmark::kMillisecond)
 
-void BM_PlanetariumPlotMethod3FarEquatorialPerspective(
-    benchmark::State& state) {
-  RunBenchmark(state, EquatorialPerspective(far));
-}
+#define PRINCIPIA_BENCHMARK_PLANETARIUM_PLOT_METHODS_POLAR_AND_EQUATORIAL( \
+    name, plotting_frame)                                                  \
+  PRINCIPIA_BENCHMARK_PLANETARIUM_PLOT_METHODS_NEAR_AND_FAR(               \
+      PolarPerspective##name, &PolarPerspective, (plotting_frame));         \
```
Comment (pleroy) [inline benchmarks/planetarium_plot_methods.cpp:345]: Bad alignment of `\`.

File: benchmarks/planetarium_plot_methods.cpp [hunk 2]
```diff
@@ -39,12 +40,17 @@ using namespace principia::integrators::_symmetric_linear_multistep_integrator;
 using namespace principia::ksp_plugin::_frames;
 using namespace principia::ksp_plugin::_planetarium;
 using namespace principia::physics::_body_centred_non_rotating_reference_frame;
+using namespace principia::physics::_body_centred_body_direction_reference_frame;  // NOLINT(whitespace/line_length)
```
Comment (pleroy) [inline benchmarks/planetarium_plot_methods.cpp:43]: I just write `NOLINT` myself.

File: benchmarks/planetarium_plot_methods.cpp [hunk 3]
```diff
@@ -58,49 +64,45 @@ constexpr Length far = 400'000 * Kilo(Metre);
...
-              Bivector<double, LeftNavigation>({0, 0, -1}))
-                  .Forget<OrthogonalMap>() *
-              Signature<Navigation, LeftNavigation>(
-                  Sign::Positive(),
-                  Sign::Positive(),
-                  DeduceSignReversingOrientation{})
-                  .Forget<OrthogonalMap>())
-          .Forget<Similarity>(),
-      focal};
+  using LeftGCRS = Frame<struct LeftGCRSTag, Arbitrary, Handedness::Left>;
```
Comment (pleroy) [inline benchmarks/planetarium_plot_methods.cpp:69]: Is there a reason for using left-handed frames?

Comment (eggrobin) [reply benchmarks/planetarium_plot_methods.cpp:69]: Probably that we do so in-game.

**Pull request merged**